### PR TITLE
Support unhashable objects in dot.

### PR DESCRIPTION
**Pull request by pelson**

### CODE REVIEW

File: dask/tests/test_dot.py [hunk 1]
```diff
@@ -46,3 +47,9 @@ def test_dot_graph():
 def test_aliases():
     g = to_networkx({'x': 1, 'y': 'x'})
     assert 'y' in g.edge['x']
+
+
+def test_np_graph():
+    g = to_networkx({'x': np.empty([1, 2])})
+    assert g.node['x']['label'] == 'x=[[ 0.  1.]]'
```
Comment (mrocklin) [inline dask/tests/test_dot.py]: Why the `[ 0. 1.]`?  Is this what happened to be in memory?  Perhaps change x to `np.array([1, 2])`?

Comment (pelson) [inline dask/tests/test_dot.py]: Yep, thanks @mrocklin - was clearly too busy thinking about racing to the finish of implementing the biggus -> dask expression graph traversal... :smile:

**Comment by pelson** [comment]
I've updated the test.

**Comment by pelson** [comment]
The py3.4 build failed with:

> dask/distributed/tests/test_scheduler.py::test_schedule

> No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.

I'm hopeful that this isn't something that I've broken. Would you mind restarting that build?

**Comment by mrocklin** [comment]
Restarted.  This is a known error and one of the reasons why I qualify dask.dataframe  :)

**Comment by mrocklin** [comment]
This looks good to me.  Merging once tests pass

**Comment by mrocklin** [comment]
In.  Thanks!

**Pull request merged**